### PR TITLE
Add type_ prop for Input/TextArea

### PR DESCRIPTION
**Pull request by c19**
`<Input type_="password" placeholder="Password" />`

### CODE REVIEW

File: bsconfig.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "package-specs": [
     {
       "module": "es6",
-      "in-source": false
+      "in-source": true
```
Comment (luisfcofv) [inline bsconfig.json:8]: Is there a reason to use `in-source`?

**Review by luisfcofv — CHANGES_REQUESTED** [review]
Thanks for your pull request. However, this project uses a different convention to generate props `[@bs.obj]`. Before merging to master, can you update your code for consistency? Thank you.

**Comment by c19** [comment]
To make debugging easier. easier to checking out compiled js.

Luis Flores <notifications@github.com> 于2018年11月2日周五 下午5:43写道：

File: src/Antd_Input.re [hunk 2]
```diff
@@ -2,28 +2,29 @@
...
-    ~style: ReactDOMRe.Style.t=?,
-    ~placeholder: string=?,
-    unit
-  ) =>
-  _ =
-  "";
+[@bs.deriving abstract]
+type jsProps = {
+  [@bs.optional] [@bs.as "type"] htmlType: string,
+  [@bs.optional] [@bs.as "name"] htmlName: string,
```
Comment (luisfcofv) [inline src/Antd_Input.re]: Why not only "name"?

File: src/Antd_Input.re [hunk 3]
```diff
@@ -2,28 +2,29 @@
...
-    ~className: string=?,
-    ~style: ReactDOMRe.Style.t=?,
-    ~placeholder: string=?,
-    unit
-  ) =>
-  _ =
-  "";
+[@bs.deriving abstract]
+type jsProps = {
+  [@bs.optional] [@bs.as "type"] htmlType: string,
```
Comment (luisfcofv) [inline src/Antd_Input.re:7]: Can you change this to "type_"? We used this name somewhere else

Comment (c19) [reply src/Antd_Input.re:7]: emmm, I changed it to htmlType because I saw `htmlType` in Button

Comment (luisfcofv) [reply src/Antd_Input.re:7]: Oh right, no problem then. I will go through the components and rename this prop later.

File: src/Antd_Input.re (hunk 2 above)
Comment (c19) [reply src/Antd_Input.re]: updated

File: src/Antd_Input.re [hunk 4]
```diff
@@ -5,7 +5,7 @@
 [@bs.deriving abstract]
 type jsProps = {
   [@bs.optional] [@bs.as "type"] htmlType: string,
-  [@bs.optional] [@bs.as "name"] htmlName: string,
+  [@bs.optional] [@bs.as "name"] name: string,
```
Comment (luisfcofv) [inline src/Antd_Input.re]: Nice one, let's remove `[@bs.as "name"]` and then I merge

Comment (c19) [reply src/Antd_Input.re]: updated

**Pull request merged**